packages/runtime/container-runtime/src/deltaScheduler.ts

```diff
@@ -57,41 +59,46 @@ export class DeltaScheduler {
         this.deltaManager.inbound.on("idle", () => { this.inboundQueueIdle(); });
     }
 
-    public batchBegin() {
+    public batchBegin(message: ISequencedDocumentMessage) {
         if (!this.processingStartTime) {
             this.processingStartTime = performance.now();
         }
+        if (this.schedulingLog === undefined && this.schedulingCount % 2000 === 0) {
```

Comment:
One thing to change here: I'd change it to === 1
Reason for that - we are not capturing very first 2000 batches, which means we have zero visibility into what happens at the beginning of container lifetime. I expect most really large batches of ops to be on boot, but this query shows zero hits for "loading" phase:

union Office_Fluid_FluidRuntime_*
| where Data_eventName contains "InboundOpsProcessingTime"
| summarize count() by Data_containerLifecycleState

I'd say - given that we are operating on number of idles / batches here, we can reduce it to % 500 or so - that would result in number of hits closer to number of times we record fluid:telemetry:OpPerf:ConnectionSpeed, which makes data a bit easier to analyze (less skew toward start or end of the session).